### ISSUE TITLE
Add windows build back

### ISSUE THREAD

**Issue by apaleyes** [EmuKit/emukit#461]
After bumping python version in #460 build on windows started failing. Error message:

```
  UPDATING build\lib.win-amd64-cpython-310\matplotlib\_version.py
  set build\lib.win-amd64-cpython-310\matplotlib\_version.py to '3.3.4'
  running build_ext
  Extracting freetype-2.6.1.tar.gz
  Building freetype in build\freetype-2.6.1
  msbuild build\freetype-2.6.1\builds\windows\vc2010\freetype.sln /t:Clean;Build /p:Configuration=Release;Platform=x64
  error: command 'msbuild' failed: None
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for matplotlib
ERROR: Could not build wheels for matplotlib, which is required to install pyproject.toml-based projects
Successfully built emukit
Failed to build matplotlib
Error: Process completed with exit code 1.
```

Definitely want to keep supporting windows, so we need to figure out what's going on

**Comment by apaleyes** [comment]
Thanks for the bump! Would you mind sharing the error you get?

**Comment by RedLenco** [comment]
Thanks for your reply! Sure, here is the output I get after running pip install emukit:

```
pip install --no-cache-dir emukit
Collecting emukit
  Downloading emukit-0.4.10.tar.gz (170 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: setuptools>=40.8.0 in c:\programdata\anaconda3\envs\fsms\lib\site-packages (from emukit) (68.0.0)
Requirement already satisfied: numpy>=1.14.5 in c:\programdata\anaconda3\envs\fsms\lib\site-packages (from emukit) (1.26.4)
Requirement already satisfied: GPy>=1.10.0 in c:\programdata\anaconda3\envs\fsms\lib\site-packages (from GPy[plotting]>=1.10.0->emukit) (1.13.2)
Collecting emcee>=2.2.1 (from emukit)
  Downloading emcee-3.1.6-py2.py3-none-any.whl.metadata (3.0 kB)
Requirement already satisfied: scipy>=1.1.0 in c:\programdata\anaconda3\envs\fsms\lib\site-packages (from emukit) (1.12.0)
Requirement already satisfied: six in c:\programdata\anaconda3\envs\fsms\lib\site-packages (from GPy>=1.10.0->GPy[plotting]>=1.10.0->emukit) (1.16.0)
Requirement already satisfied: paramz>=0.9.6 in c:\programdata\anaconda3\envs\fsms\lib\site-packages (from GPy>=1.10.0->GPy[plotting]>=1.10.0->emukit) (0.9.6)
Requirement already satisfied: cython>=0.29 in c:\programdata\anaconda3\envs\fsms\lib\site-packages (from GPy>=1.10.0->GPy[plotting]>=1.10.0->emukit) (3.0.11)
Collecting matplotlib==3.3.4 (from GPy[plotting]>=1.10.0->emukit)
  Downloading matplotlib-3.3.4.tar.gz (37.9 MB)
     ---------------------------------------- 37.9/37.9 MB 109.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [16 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\xxxxxxx\AppData\Local\Temp\12\pip-install-_lsw6f11\matplotlib_995e1cd86bcc42cd85620c08d5394da5\setup.py", line 52, in <module>
          __version__ = versioneer.get_version()
                        ^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\xxxxxxx\AppData\Local\Temp\12\pip-install-_lsw6f11\matplotlib_995e1cd86bcc42cd85620c08d5394da5\versioneer.py", line 1410, in get_version
          return get_versions()["version"]
                 ^^^^^^^^^^^^^^
        File "C:\Users\xxxxxxx\AppData\Local\Temp\12\pip-install-_lsw6f11\matplotlib_995e1cd86bcc42cd85620c08d5394da5\versioneer.py", line 1344, in get_versions
          cfg = get_config_from_root(root)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\xxxxxxx\AppData\Local\Temp\12\pip-install-_lsw6f11\matplotlib_995e1cd86bcc42cd85620c08d5394da5\versioneer.py", line 401, in get_config_from_root
          parser = configparser.SafeConfigParser()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

So if I understand the output correctly this is the error:
 ```
python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [16 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\xxxxxxx\AppData\Local\Temp\12\pip-install-_lsw6f11\matplotlib_995e1cd86bcc42cd85620c08d5394da5\setup.py", line 52, in <module>
          __version__ = versioneer.get_version()
                        ^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\xxxxxxx\AppData\Local\Temp\12\pip-install-_lsw6f11\matplotlib_995e1cd86bcc42cd85620c08d5394da5\versioneer.py", line 1410, in get_version
          return get_versions()["version"]
                 ^^^^^^^^^^^^^^
        File "C:\Users\xxxxxxx\AppData\Local\Temp\12\pip-install-_lsw6f11\matplotlib_995e1cd86bcc42cd85620c08d5394da5\versioneer.py", line 1344, in get_versions
          cfg = get_config_from_root(root)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\xxxxxxx\AppData\Local\Temp\12\pip-install-_lsw6f11\matplotlib_995e1cd86bcc42cd85620c08d5394da5\versioneer.py", line 401, in get_config_from_root
          parser = configparser.SafeConfigParser()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
```

Any recommendations are greatly appreciated. Thanks in advance!

**Comment by apaleyes** [comment]
Thanks, we'll have a look. Can't promise any specific timeline for the fix though.

General recommendation is to see if other people faced same problem. It is clear that the problem isn't emukit but matplotlib. So it's always worth searching internet for the error text. For example, here are top two links (out of many) I found, which may or may not help you too:
* https://stackoverflow.com/questions/72616748/python-setup-py-egg-info-did-not-run-successfully-exit-code-1
* https://stackoverflow.com/questions/76129688/why-is-pip-install-gym-failing-with-python-setup-py-egg-info-did-not-run-succ

**Comment by apaleyes** [comment]
An update here: fixes probably won't help. Apparently GPy (our dependency) hardcodes certain mpl version.

To unblock yourself, consider pulling the branch behind the above PR or, once it's merged, main. We will also look into releasing new version into pip soon.

**Comment by apaleyes** [comment]
The fix was released as in 0.4.11. Thanks again, hopefully it works for on your machine now @RedLenco
